src/main/scala/com/exasol/cloudetl/kafka/consumer/KafkaRecordConsumer.scala

```diff
@@ -57,11 +58,64 @@ class KafkaRecordConsumer(
         )
       } while (shouldContinue(recordOffset, recordCount, totalRecordCount))
     } catch {
+      case exception: IllegalStateException =>
+        throw new KafkaConnectorException(
+          ExaError
+            .messageBuilder("E-KCE-20")
+            .message("Error consuming Kafka topic {{TOPIC}} data. ", topic)
+            .message("Consumer is not subscribed to any topic or assigned any partition.")
+            .mitigation("Please check that the Kafka topic is available and valid.")
+            .toString(),
+          exception
+        )
+      case exception: InvalidTopicException =>
+        throw new KafkaConnectorException(
+          ExaError
+            .messageBuilder("E-KCE-21")
+            .message(generalErrorMessage(), topic)
+            .message("Provided topic is not valid.")
+            .mitigation("Please check that the Kafka topic is available and valid.")
+            .toString(),
+          exception
+        )
+      case exception: TimeoutException =>
+        throw new KafkaConnectorException(
+          ExaError
+            .messageBuilder("E-KCE-22")
+            .message(generalErrorMessage(), topic)
+            .message("Timeout trying to connect to Kafka brokers.")
+            .mitigation(
+              "Please ensure that there is network connection between Kafka brokers and Exasol datanode." +
+                "Similarly check that Kafka advertised listeners are reachable from Exasol cluster."
```

Comment:
Add one more mitigation instead of using +. They will be rendered into a list nicely